content/docs/specification/content.md

```diff
@@ -43,151 +43,193 @@ Further links about URL encoding:
 
-**Redirect URL**  
+### Path Sanitization
 
-* Key: **to**
-* Recommended
-* Default: Fallback to **www** or **redirect** config
-* Permitted values: "absolute|relative URL"
-* Example: "to=https://example.com"
-* Example: "to=/example/"
+To use the request's path for generating the zone address we need normalize the path into allowed DNS chars. For example we need to replace `.`(dot) into `-`(dash) to keep each match in its own zone. You can visit [RFC1034](https://tools.ietf.org/html/rfc1034) for more information about the allowed chars.
 
-**Root Redirect URL**  
+After this step we run our default regex or the custom regex from `re=` field on the request's path. If `from=` is not available we first revert order of matches from the expression then join them with request's host to generate the next zone in the chain. But if `from=` is availabe, we use the given order instead of reverting.
 
-* Key: **root**
-* Recommended
-* Default: Fallback to **www** or **redirect** config
-* Permitted values: "absolute|relative URL"
-* Example: "to=https://example.com"
-* Example: "to=/example/"
+## Record Fields
+
+**Type**
+
+- Key: **type**
+- Mandatory
+- Permitted values: "path"
+- Example: "type=path"
+
+**Version**
 
-**Simplified Regex**  
+- Key: **v**
+- Mandatory
+- Permitted values: "txtv0"
+- Example: "v=txtv0"
 
-* Key: **from**
-* Permitted values: "simplified regex"
-* Example: "from=/$2/$1/$3"
+**Redirect URL**
 
-**Regex**  
+- Key: **to**
+- Recommended
+- Default: Fallback to **www** or **redirect** config
+- Permitted values: "absolute|relative URL"
+- Example: "to=https://example.com"
+- Example: "to=/example/"
 
-* Key: **re**
-* Permitted values: "regex/ordered regex"
-* Note: "Named regexes can be used to reorder the results. a > b > b1 > b2"
-* Example: "`from=\\?query=(?P<a>[^&]+)\\&more=(?P<b>[^&]+)`"
+**Root Redirect URL**
+
+- Key: **root**
+- Recommended
+- Default: Fallback to **www** or **redirect** config
+- Permitted values: "absolute|relative URL"
+- Example: "to=https://example.com"
+- Example: "to=/example/"
+
+**Simplified Regex**
+
+- Key: **from**
+- Permitted values: "simplified regex"
+- Example: "from=/$2/$1/\$3"
+
+**Regex**
+
+- Key: **re**
+- Permitted values: "regex/ordered regex"
+- Note: "Named regexes can be used to reorder the results. a > b > b1 > b2"
+- Example: "`from=\\?query=(?P<a>[^&]+)\\&more=(?P<b>[^&]+)`"
 
 ---
+
 # Dockerv2 Type
-**Type**  
 
-* Key: **type**
-* Mandatory
-* Permitted values: "dockerv2"
-* Example: "type=dockerv2"
+## Description
 
-**Version**  
+To redirect request from Docker CLI to a specific docker image or a docker registry you can use `dockerv2` type. The difference between a Docker CLI request and a normal request is that Docker CLI sends a couple of request with different paths to fetch some information like image tags, manifests and etc. about the image. To handle these requests we use docker's v2 API to fetch the data for each request.
```

Comment:
Not sure about redirecting. That's done in the backend. For the user it's probably more about controlling the endpoint or have a vanity url. How about:
```suggestion
The `dockerv2` type enables vanity URLs for your container images. Using your own URl for container images enables easier switching between your backend infrastructure/storage and adds more control.
The type implements the Docker registry API v2 and redirects requests for both metadata and image blobs to the backend storage. All data besides the redirect is therefore served by your underlying infrastructure such as Google Container Registry (gcr.io).
```

Probably a general issue is that we are conflating the specification a bit with user docs. Ok for now but in the long term with a new implementation we might need to switch the language from user specific to implementation specific.

So instead of "can be used for" to "the type needs to implement the docker registry api v2 as documented here".